base/ltmeta.dtx

```diff
@@ -0,0 +1,189 @@
+% \end{abstract}
+%
+% \tableofcontents
+%
+%
+% \section{Introduction}
+%
+%
+%    In the past there was no dedicated location to declare settings
+%    concerning a document as a whole. Settings are placed somewhere
+%    in the preamble or with the class options or even with some
+%    package options.  For some settings this can be too late, for
+%    example the pdf version can no longer be changed if a package has
+%    used code which already opened the PDF.
+%
+%    \cs{DocumentMetadata} as a new command unifies such settings in
+%    one place.  It must be used before \cs{documentclass} but can be
+%    issued more than once there.
+%
+%    At the moment most of the code run by \cs{DocumentMetadata} is
+%    external to the format and subject to change. This includes the
+%    supported key/values.
+%
+%    For that reason all that happens right now in the format is to
+%    look for suitable support files and if found, to redirect the
+%    processing to them.
+%
+% \subsection{\cs{DocumentMetadata}}
+%
+% \begin{function}{\DocumentMetadata}
+% \begin{syntax}
+%  \cs{DocumentMetadata}\Arg{key-value list}
+% \end{syntax}
+% \end{function}
+%
+%
+%
+%    The keys defined for \cs{DocumentMetadata}
+%    currently allow to set the PDF version, to set the PDF \texttt{/Lang},
+%    to uncompress a PDF, to set the language and to declare a few PDF standards
+%    and some color profiles.
+%
+%    \cs{DocumentMetadata} is also used to
+%    activate the new PDF management code and it loads
+%    a number of required files for the PDF management code.
+%    As this forces the loading of the backend files, a backend
+%    which can't be detected automatically like |dvipdfmx|,
+%    must be set in the first \cs{DocumentMetadata} call (if there is more than one).
+%
+%    The full set of keys currently supported is documented elsewhere
+%    for now.
+%
+%
+% \StopEventually{\setlength\IndexMin{200pt}  \PrintIndex  }
+%
+%
+% \section{The Implementation}
+%
+%
+%
+%    \begin{macrocode}
+%<*2ekernel|latexrelease>
+%    \end{macrocode}
+%    Not needed yet but \ldots
+%    \begin{macrocode}
+%\ExplSyntaxOn
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+%<latexrelease>\NewModuleRelease{2021/06/01}{ltmeta}
+%<latexrelease>                 {Document Metadata handling}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\newcommand\DocumentMetadata{%
+  \InputIfFileExists{documentmetadata-support.ltx}%
+     {}%
+%    \end{macrocode}
+%    The above file is changing \cs{DocumentMetadata} to a suitable
+%    definition (or so we hope), so that we can try again --- if not
+%    tough.
+% 
+%    If the file can't be found we say so and carry on without it.
+%    \begin{macrocode}
+     {%
+       \@latex@warning@no@line{No support files for
+         \noexpand\DocumentMetadata found.\MessageBreak
+         Is the 'documentmetadata-support' bundle installed?%
+         \MessageBreak\MessageBreak
+         The declaration is ignored}%
```

Comment:
Not sure about this, but I think I would prefer an error here. `\DocumentMetadata`, especially since it loads PDF managment and other functionality, might easily change the behavior of the document and therefore ignoring it should not be hidden.

Reply:
you are probably right. Changed.